pkg/controller/job/job_controller.go

```diff
@@ -688,6 +688,14 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 	// make a copy so we don't mutate the shared cache
 	job := *sharedJob.DeepCopy()
 
+	// Let us queue the job back accounting to the activeDeadlineSeconds if the item is not marked to be forgotten
+	defer func(key string, job *batch.Job) {
```

Comment:
First, we'd better put this after `IsJobFinished` check, so that we won't AddAfter a finished job.
Second, why not just put this logic into the existing defer in L716-722 ? You don't have to define the arguments of defer func, for you can directly use them in the function by closure.

Reply:
Ack. Changed accordingly.

Reply:
The reason to keep a different defer was because of what was in  https://github.com/kubernetes/kubernetes/blob/4c125cca5621501f1ace2d362eaed297fda0ab37/pkg/controller/job/job_controller.go#L704-L707

However, I think it should be fine if we don't re-queue the event there anyway since the `syncJob` itself is not going to run for unknown `CompletionMode`.